SpatialGDK/Source/Private/Interop/SpatialSender.cpp

```diff
@@ -597,16 +597,16 @@ Worker_ComponentUpdate USpatialSender::CreateMulticastUpdate(UObject* TargetObje
 
 	OutEntityId = TargetObjectRef.Entity;
 
-	TSet<const UObject*> UnresolvedObjects;
+	TSet<TWeakObjectPtr<const UObject>> UnresolvedObjects;
 	FSpatialNetBitWriter PayloadWriter(PackageMap, UnresolvedObjects);
 
 	TSharedPtr<FRepLayout> RepLayout = NetDriver->GetFunctionRepLayout(Function);
 	RepLayout_SendPropertiesForRPC(*RepLayout, PayloadWriter, Parameters);
 
-	for (const UObject* Object : UnresolvedObjects)
+	for (TWeakObjectPtr<const UObject> Object : UnresolvedObjects)
 	{
 		// Take the first unresolved object
-		OutUnresolvedObject = Object;
+		OutUnresolvedObject = Object.Get();
```

Comment:
Perhaps we should change OutUnresolvedObject to also be a TWeakObjectPtr so it's usages are also protected somewhat.